src/Components/ImagesTable/ImagesTable.tsx

```diff
@@ -153,6 +171,81 @@ const ImagesTable = () => {
     );
   }
 
+  let composes = data?.data;
+  if (selectedBlueprintId && blueprintVersionFilter === 'latest') {
+    composes = composes?.filter((compose) => {
+      return compose.blueprint_version === selectedBlueprintVersion;
+    });
+  }
+  const itemCount = data?.meta.count || 0;
+
+  if (isOnPremise === 'cockpit') {
```

Comment:
For a better maintainable code, env logic should be separated from component logic as far as we can. I think it would be better to use abstract environment specific logic, separating the component logic from the env, by using dedicated config files for each env (i.e for API specification),  HOC (high order components), React Context for propagating the `isOnPrem`, dynamic importing and more.